component/src/main/java/com/alfresco/content/component/SearchUserComponentSheet.kt

```diff
@@ -99,10 +100,30 @@ class SearchUserComponentSheet : BottomSheetDialogFragment(), MavericksView {
 
     private fun setSearchQuery(query: String) {
         val term = cleanupSearchQuery(query)
+        if (!viewModel.searchByName) {
+            if (isValidString(term))
+                executeSearch(term)
+        } else executeSearch(term)
+    }
+
+    private fun executeSearch(term: String) {
         scrollToTop()
         viewModel.setSearchQuery(term)
     }
 
+    private fun isValidString(str: String): Boolean {
+        val emailAddressPattern = Pattern.compile(
```

Comment:
Do we not have email validator codes in common?
If not then make it common for future reference.